### PR TITLE
feat: add to Watch Later

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
Add & delete watch later from 

- Home Page
- Single Video Page

Screenshot (150)
Screenshot (151)

### CODE REVIEW

File: src/components/Aside/Aside.jsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const Aside=()=>{
         <div className="aside-tab flex-row"
         // onClick={sideBarHandler}
```
Comment (anuj-kumary) [inline src/components/Aside/Aside.jsx:23]: Remove unwanted comment

File: src/components/WatchLaterModal/WatchLaterModal.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+           <div className="watch-modal-option flex-row"
+           onClick={addToWatchlater}
+           >
+            {isInWatchLater?
+            <div className="remove-watch flex-row">
+            <i class="far fa-trash-alt"></i>
+            <p>Remove From Watch Later</p>
+            </div>:
+            <>
+            <i class="far fa-clock"></i>
```
Comment (anuj-kumary) [inline src/components/WatchLaterModal/WatchLaterModal.jsx]: Change class to className

File: src/utils/WatchLater.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const removeVideoFromWatchLater = async (id, token, dispatch) => {
+  try {
+    const response = await removeWatchLaterServices(id, token);
+    if (response.status === 200 || response.status === 201)
+      dispatch({
+        type: ACTION_TYPE.REMOVE_WATCH,
+        payload: response.data.watchlater,
+      });
+  } catch (error) {
+    console.error();
```
Comment (anuj-kumary) [inline src/utils/WatchLater.js:39]: pass error in `console.error()`

**Review by anuj-kumary — APPROVED** [review]
- The overall code looks good easy to read
- Good to merge

File: src/components/VIdeoCard/VideoCard.jsx [hunk 4]
```diff
@@ -1,10 +1,23 @@
 import "./VideoCard.css";
+import {useState} from "react";
+import { useAuth } from "../../contexts/auth-context";
```
Comment (ramalingaa) [inline src/components/VIdeoCard/VideoCard.jsx]: You can have one index for context files here

File: src/components/VIdeoCard/VideoCard.jsx [hunk 5]
```diff
@@ -1,10 +1,23 @@
...
+import { useData } from "../../contexts/data-context";
+import {WatchLaterModal} from "../WatchLaterModal/WatchLaterModal"
 import { useNavigate } from "react-router-dom";
+import { WatchLaterHandler } from "../../utils/WatchLater";
```
Comment (ramalingaa) [inline src/components/VIdeoCard/VideoCard.jsx:7]: For components as well

**Review by ramalingaa — APPROVED** [review]
Code strcure is good and logic is working. PR looks good to merge

File: src/components/VIdeoCard/VideoCard.jsx [hunk 6]
```diff
@@ -1,10 +1,23 @@
...
 
 export const VideoCard=({item})=>{
     const {image,title,creator,date,_id}=item || {};
     const navigate=useNavigate();
+    const {token}= useAuth();
+    const {state,dispatch}= useData();
+    const [modal,setModal]= useState(false);
+    const isInWatchLater=state.watchlater.some((ele)=>ele._id ===_id)
+    const modalHandler=()=>{
```
Comment (vinc3nati) [inline src/components/VIdeoCard/VideoCard.jsx:16]: trim the fat here

**Review by vinc3nati — APPROVED** [review]

File: src/components/Aside/Aside.jsx (hunk 1 above)
Comment (PeeyushPrashant) [reply src/components/Aside/Aside.jsx:23]: these will be used later on in the project

**Pull request merged**